cpp/include/cudf/copying.hpp

```diff
@@ -485,7 +485,21 @@ std::vector<table_view> split(table_view const& input, std::vector<size_type> co
  * table metadata and one on device which contains the table data.
  */
 struct packed_columns {
-  std::unique_ptr<std::vector<uint8_t>> metadata;
+  /**
+   * @brief Host-side metadata buffer used for reconstructing columns via unpack.
+   *
+   * @ingroup copy_split
+   */
+  struct metadata {
+    metadata(std::vector<uint8_t>&& v) : data_(v) {}
```

Comment:
Careful, an r-value ref is itself an l-value, so it still needs to be `move`d. 
```suggestion
    metadata(std::vector<uint8_t>&& v) : data_(std::move(v)) {}
```